src/components/EncounterDetails/EncounterDetails.jsx

```diff
@@ -1,182 +1,292 @@
+                </div>
+                <div>
+                  <h3>Darkvision:</h3>
+                  <p>{monster.details && monster.details.senses.darkvision}</p>
+                </div>
+                <div>
+                  <h3>Passive Perception:</h3>
+                  <p>{monster.details && monster.details.senses.passive_perception}</p>
+                </div>
+              </section>
+              <section className='proficiencies'>{proficienciesArray}</section>
+              <section className='special-abilities'>{specialAbilitiesArray}</section>
+              <section className='standard-actions'>{standardActionsArray}</section>
+              <section className='legendary-actions'>{legendaryActionsArray}</section>
+            </details>
+          );
+        });
+
+        setSelectedMonsters(monsters);
+        setLoading(false);
+      } 
+      fetchMonsterDetails();
     }
-  }, [encounters]);
+  }, [selectedEncounter]);
 
   return (
     <div className='EncounterDetails'>
-      <section className='main-details'>
-        <section className='non-monster base-box'>
-          <section className='encounter-header'>
-            <h2 className='encounter-name'>{encounters[0] && encounters[0].encounter_name}</h2>
-            <div className='party-stats'>
-              <h3>Party Size:</h3>
-              <p>{encounters[0] && encounters[0].party_size}</p>
-              <h3>Party Level:</h3>
-              <p>{encounters[0] && encounters[0].party_level}</p>
-            </div>
+      {selectedEncounter ? (
+        <section className='main-details'>
+          <section className='non-monster base-box'>
+            <section className='encounter-header'>
+              <h2 className='encounter-name'>{selectedEncounter.encounterName}</h2>
+              <div className='party-stats'>
+                <h3>Party Size:</h3>
+                <p>{selectedEncounter.partySize}</p>
+                <h3>Party Level:</h3>
+                <p>{selectedEncounter && selectedEncounter.partyLevel}</p>
+              </div>
+            </section>
+            <section className='encounter-desc'>
+              <h3>Short Summary:</h3>
+              <p>{selectedEncounter.summary}</p>
+              <h3>Encounter Description:</h3>
+              <p>{selectedEncounter.description}</p>
+              <h3>Treasure and Rewards:</h3>
+              <p>{selectedEncounter.treasure}</p>
+            </section>
           </section>
-          <section className='encounter-desc'>
-            <h3>Short Summary:</h3>
-            <p>{encounters[0] && encounters[0].summary}</p>
-            <h3>Encounter Description:</h3>
-            <p>{encounters[0] && encounters[0].description}</p>
-            <h3>Treasure and Rewards:</h3>
-            <p>{encounters[0] && encounters[0].treasure}</p>
+          <section className='base-box encounter-foes'>
+            <h2>Your Enemies</h2>
+            {loading ? (
```

Comment:
This will be handy with how beefy some of these datasets can get, thank you for putting this in!